docs/data-ai/data/data-pipelines.md

```diff
@@ -0,0 +1,756 @@
+{{% tab name="Python" %}}
+
+Use [`DataClient.CreateDataPipeline`](/dev/reference/apis/data-client/#createdatapipeline) to define a new pipeline:
+
+```python
+data_client = DataClient.from_api_key(
+    api_key="<api-key>",
+    api_key_id="<api-key-id>"
+)
+
+request = data_client.create_data_pipeline(
+    organization_id="<org-id>",
+    name="hourly-temp-average",
+    mql_binary=[
+        bson.encode({"$match": {"component_name": "temperature-sensor"}}),
+        bson.encode({
+            "$group": {
+                "_id": "$location_id",
+                "avg_temp": {"$avg": "$data.readings.temperature"},
+                "count": {"$sum": 1}
+            }
+        })
+    ],
+    schedule="0 * * * *"  # Run hourly
+)
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+Use [`DataClient.CreateDataPipeline`](https://pkg.go.dev/go.viam.com/rdk/app#DataClient.CreateDataPipeline) to define a new pipeline:
+
+```go
+client, err := utils.NewViamClient(context.Background(), utils.WithAPIKey("<api-key>", "<api-key-id>"))
+if err != nil {
+    panic(err)
+}
+defer client.Close()
+
+dataClient := client.DataClient()
+
+pipeline := [][]byte{
+    bson.Marshal(bson.M{"$match": bson.M{"component_name": "temperature-sensor"}}),
+    bson.Marshal(bson.M{
+        "$group": bson.M{
+            "_id": "$location_id",
+            "avg_temp": bson.M{"$avg": "$data.readings.temperature"},
+            "count": bson.M{"$sum": 1},
+        },
+    }),
+}
+
+resp, err := dataClient.CreateDataPipeline(context.Background(), &datapb.CreateDataPipelineRequest{
+    OrganizationId: "<org-id>",
+    Name: "hourly-temp-average",
+    MqlBinary: pipeline,
+    Schedule: "0 * * * *", // Run hourly
+})
+```
+
+{{% /tab %}}
+{{% tab name="TypeScript" %}}
+
+Use [`dataClient.CreateDataPipeline`](/dev/reference/apis/data-client/#createdatapipeline) to define a new pipeline:
+
+```typescript
+const apiKey = "<api-key>";
+const apiKeyID = "<api-key-id>";
+
+const client = await createViamClient({
+  credential: {
+    type: "api-key",
+    payload: { key: apiKey, keyId: apiKeyID },
+  },
+});
+
+const dataClient = client.dataClient;
+
+const pipeline = [
+  BSON.serialize({ $match: { component_name: "temperature-sensor" } }),
+  BSON.serialize({
+    $group: {
+      _id: "$location_id",
+      avg_temp: { $avg: "$data.readings.temperature" },
+      count: { $sum: 1 },
+    },
+  }),
+];
+
+const response = await dataClient.createDataPipeline({
+  organizationId: "<org-id>",
+  name: "hourly-temp-average",
+  mqlBinary: pipeline,
+  schedule: "0 * * * *", // Run hourly
+});
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+To create a pipeline that reads data from the [hot data store](/data-ai/data/hot-data-store/), specify a `dataSourceType` in your pipeline configuration.
+
+#### Schedule format
+
+To create a schedule for your pipeline, specify a [cron expression](https://en.wikipedia.org/wiki/Cron) in UTC.
+The schedule determines both execution frequency and input time window.
```

Comment:
Will readers know what "input time window means"?

Reply:
Good point, changed to 'query time range' to make this easier to grok. If you still think that's confusing, I can add a more in-depth explanation.